cmd/ctl/pkg/renew/renew.go

```diff
@@ -0,0 +1,235 @@
+	cmapi "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1alpha2"
+	cmmeta "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
+	cmclient "github.com/jetstack/cert-manager/pkg/client/clientset/versioned"
+)
+
+var (
+	long = templates.LongDesc(i18n.T(`
+Mark cert-manager Certificate resources for manual renewal.`))
+
+	example = templates.Examples(i18n.T(`
+# Renew the Certificates named 'my-app' and 'vault' in the current context namespace.
+ctl renew my-app vault
+
+# Renew all Certificates in the 'kube-system' namespace.
+ctl renew --namespace kube-system --all
+
+# Renew all Certificates in all namespaces, provided those Certificates have the label 'app=my-service
+ctl renew --all-namespaces -l app=my-service`))
+)
+
+// Options is a struct to support renew command
+type Options struct {
+	CMClient   cmclient.Interface
+	RESTConfig *restclient.Config
+
+	// The Namespace that the Certificate to be renewed resided in.
+	// This flag registration is handled by cmdutil.Factory
+	Namespace     string
+	LabelSelector string
+	All           bool
+	AllNamespaces bool
+
+	genericclioptions.IOStreams
+}
+
+// NewOptions returns initialized Options
+func NewOptions(ioStreams genericclioptions.IOStreams) *Options {
+	return &Options{
+		IOStreams: ioStreams,
+	}
+}
+
+// NewCmdRenew returns a cobra command for renewing Certificates
+func NewCmdRenew(ioStreams genericclioptions.IOStreams) *cobra.Command {
+	o := NewOptions(ioStreams)
+	var factory cmdutil.Factory
+
+	cmd := &cobra.Command{
+		Use:     "renew",
+		Short:   "Mark a Certificate for manual renewal",
+		Long:    long,
+		Example: example,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(factory))
+			cmdutil.CheckErr(o.Validate(cmd, args))
+			cmdutil.CheckErr(o.Run(args))
+		},
+	}
+
+	cmd.Flags().StringVarP(&o.LabelSelector, "selector", "l", o.LabelSelector, "Selector (label query) to filter on, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2)")
+	cmd.Flags().BoolVarP(&o.AllNamespaces, "all-namespaces", "A", o.AllNamespaces, "If present, mark Certificates across namespaces for manual renewal. Namespace in current context is ignored even if specified with --namespace.")
+	cmd.Flags().BoolVar(&o.All, "all", o.All, "Renew all Certificates in the given Namespace, or all namespaces with --all-namespaces enabled.")
+
+	kubeConfigFlags := genericclioptions.NewConfigFlags(true)
+	kubeConfigFlags.AddFlags(cmd.PersistentFlags())
+	matchVersionKubeConfigFlags := cmdutil.NewMatchVersionFlags(kubeConfigFlags)
+	matchVersionKubeConfigFlags.AddFlags(cmd.PersistentFlags())
+	factory = cmdutil.NewFactory(matchVersionKubeConfigFlags)
+
+	cmd.Flags().AddGoFlagSet(flag.CommandLine)
+	flag.CommandLine.Parse([]string{})
+	fakefs := flag.NewFlagSet("fake", flag.ExitOnError)
+	klog.InitFlags(fakefs)
+	if err := fakefs.Parse([]string{"-logtostderr=false"}); err != nil {
+		fmt.Fprintf(os.Stderr, "%s\n", err)
+		os.Exit(1)
+	}
```

Comment:
All of this up to line 103 should not be present (i.e. https://github.com/jetstack/cert-manager/pull/2845/files#diff-cb72cb254bdf97835bab48594e5c84ecR103-R116) - this should all be handled at the top level